pkg/console/controllers/managedclusters/controller.go

```diff
@@ -229,18 +229,21 @@ func (c *ManagedClusterController) SyncManagedClusterList(ctx context.Context) (
 		valid = append(valid, managedCluster)
 	}
 
-	return valid, nil, ""
+	return valid, "", nil
 }
 
-func (c *ManagedClusterController) SyncOAuthClientManagedClusterViews(ctx context.Context, operatorConfig *operatorv1.Console, managedClusters []clusterv1.ManagedCluster) ([]*unstructured.Unstructured, error, string) {
+func (c *ManagedClusterController) SyncOAuthClientManagedClusterViews(ctx context.Context, operatorConfig *operatorv1.Console, managedClusters []clusterv1.ManagedCluster) ([]*unstructured.Unstructured, string, error) {
 	errs := []string{}
 	mcvs := []*unstructured.Unstructured{}
 	for _, managedCluster := range managedClusters {
-		required := managedclusterviewsub.DefaultOAuthClientView(managedCluster.Name)
-		mcv, err := c.dynamicClient.Resource(api.ManagedClusterViewGroupVersionResource).Namespace(managedCluster.Name).Create(ctx, required, metav1.CreateOptions{})
-		if err != nil && apierrors.IsAlreadyExists(err) {
-			mcvName, _ := managedclusterviewsub.GetName(required)
-			mcv, err = c.dynamicClient.Resource(api.ManagedClusterViewGroupVersionResource).Namespace(managedCluster.Name).Get(ctx, mcvName, metav1.GetOptions{})
+		mcv, err := c.dynamicClient.Resource(api.ManagedClusterViewGroupVersionResource).Namespace(managedCluster.Name).Get(ctx, api.OAuthClientManagedClusterViewName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			required, err := managedclusterviewsub.DefaultOAuthClientView(managedCluster.Name)
+			if err != nil {
+				errs = append(errs, fmt.Sprintf("Error initializing oauth client ManagedClusterView for cluster %s: %v", managedCluster.Name, err))
+				continue
+			}
+			mcv, err = c.dynamicClient.Resource(api.ManagedClusterViewGroupVersionResource).Namespace(managedCluster.Name).Create(ctx, required, metav1.CreateOptions{})
```

Comment:
Since we have to do error checking on initializing the required MCV, the logic needed to be switched around here. Instead of trying to create the resource, then handling the case where it already exists, we check to see if it already exists, then create it if not.